supplier_product/models.py

```diff
@@ -0,0 +1,35 @@
+from django.db import models
+from django.core.validators import MinValueValidator
+from product.models import Product
+from supplier.models import Supplier
+
+
+class SupplierProduct(models.Model):
+    suppler_product_id = models.IntegerField(primary_key=True)
+    qr_code = models.ForeignKey(Product, on_delete=models.CASCADE)
+    user_name = models.ForeignKey(Supplier, on_delete=models.CASCADE)
+    price = models.IntegerField(validators=[MinValueValidator(1)])
+    quantity = models.IntegerField()
+
+    class Meta:
+        unique_together = ('user_name', 'qr_code')
+
+    @staticmethod
+    def save_sup_product(sup_prod):
+        SupplierProduct.full_clean(sup_prod)
+        sup_prod.save()
+
+    @staticmethod
+    def delete_sup_product(self):
```

Comment:
A static method cannot accept `self` as an argument. 
Static methods are methods that are bound to a class rather than its object. You can read more about it [here](https://www.tutorialsteacher.com/python/staticmethod-decorator).
There is no need for this method here, please delete it and use django `delete()` method.